easydock/vina_dock.py

```diff
@@ -75,51 +75,63 @@ def mol_dock(mol, config):
-        subprocess.run(cmd, shell=True, check=True, capture_output=True, text=True)  # this will trigger CalledProcessError and skip next lines)
-        dock_time = round(timeit.default_timer() - start_time, 1)
-
-        with open(output_fname) as f:
-            res = f.read()
-            if res:
-                res = json.loads(res)
-                mol_block = pdbqt2molblock(res['poses'].split('MODEL')[1], mol, mol_id)
-                output = {'docking_score': res['docking_score'],
-                          'pdb_block': res['poses'],
-                          'mol_block': mol_block,
-                          'dock_time': dock_time}
-
-    except subprocess.CalledProcessError as e:
-        sys.stderr.write(f'Error caused by docking of {mol_id}\n')
-        sys.stderr.write(str(e) + '\n')
-        sys.stderr.write('STDERR output:\n')
-        sys.stderr.write(e.stderr + '\n')
-        sys.stderr.flush()
-        output = None
-
-    finally:
-        os.close(output_fd)
-        os.close(ligand_fd)
-        os.unlink(ligand_fname)
-        os.unlink(output_fname)
+    dock_output_conformer_list = []
+    start_time = timeit.default_timer()
+    for ligand_pdbqt in ligand_pdbqt_list:
+        output_fd, output_fname = tempfile.mkstemp(suffix='_output.json', text=True)
+        ligand_fd, ligand_fname = tempfile.mkstemp(suffix='_ligand.pdbqt', text=True)
+
+        try:
+            with open(ligand_fname, 'wt') as f:
+                f.write(ligand_pdbqt)
+
+            p = os.path.realpath(__file__)
+            python_exec = sys.executable
+            cmd = f'{python_exec} {os.path.dirname(p)}/vina_dock_cli.py -l {ligand_fname} -p {config["protein"]} ' \
+                f'-o {output_fname} --center {" ".join(map(str, config["center"]))} ' \
+                f'--box_size {" ".join(map(str, config["box_size"]))} ' \
+                f'-e {config["exhaustiveness"]} --seed {config["seed"]} --nposes {config["n_poses"]} -c {config["ncpu"]}'
+            subprocess.run(cmd, shell=True, check=True, capture_output=True, text=True)  # this will trigger CalledProcessError and skip next lines)
+
+            with open(output_fname) as f:
+                res = f.read()
+                if res:
+                    res = json.loads(res)
+                    mol_block = pdbqt2molblock(res['poses'].split('MODEL')[1], mol, mol_id)
+                    output = {'docking_score': res['docking_score'],
+                            'pdb_block': res['poses'],
+                            'mol_block': mol_block}
+                    
+                    dock_output_conformer_list.append(output)
+            
+        except subprocess.CalledProcessError as e:
+            sys.stderr.write(f'Error caused by docking of {mol_id}\n')
+            sys.stderr.write(str(e) + '\n')
+            sys.stderr.write('STDERR output:\n')
+            sys.stderr.write(e.stderr + '\n')
+            sys.stderr.flush()
+            output = None
```

Comment:
This line is already not necessary and can be deleted. If docking was unsuccessful it will not contribute to the `output_list`